src/fides/api/ctl/sql_models.py

```diff
@@ -186,6 +186,26 @@ class DataUse(Base, FidesBase):
     legitimate_interest_impact_assessment = Column(String, nullable=True)
     is_default = Column(BOOLEAN, default=False)
 
+    @staticmethod
+    def get_parent_uses(data_use_key: str) -> Set[str]:
+        """
+        Utility method to traverse "up" the taxonomy hierarchy and unpack
+        a given data use fides key into a set of fides keys that include its
+        parent fides keys.
+
+        The utility takes a fides key string input to make the method more applicable -
+        since in many spots of our application we do not have a true `DataUse` instance,
+        just a "soft" reference to its fides key.
+
+        Example inputs and outputs:
+            - `a.b.c` --> [`a.b.c`, `a.b`, `a`]
+            - `a` --> [`a`]
+        """
+        parent_uses = {data_use_key}
+        while data_use_key := data_use_key.rpartition(".")[0]:
```

Comment:
ah I hadn't see `:=` used before

Reply:
😄 there've been a few times where i thought it would make sense but it didn't pan out. i _think_ it works pretty well here? there may have still been a cleaner way of doing this, but figured this was good enough for now!